src/action.js

```diff
@@ -255,19 +280,61 @@ async function setStatus(pullRequest, state, description) {
   });
 }
 
+async function generateReleaseNotes(pullRequest) {
+  const tag = getTagName(pullRequest);
+
+  var previous_tag_name;
+  try {
+    const latest_release = await client.rest.repos.getLatestRelease({
+      owner: github.context.repo.owner,
+      repo: github.context.repo.repo,
+    });
+    previous_tag_name = latest_release.data.tag_name;
+    core.info(
+      "Generating release-notes relative to release " + previous_tag_name + ".."
+    );
+  } catch (error) {
+    previous_tag_name = "";
+    core.info("Generating release-notes relative to start of repository..");
+  }
+
+  try {
+    const response = await client.request(
+      "POST /repos/{owner}/{repo}/releases/generate-notes",
+      {
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
+        tag_name: tag,
+        target_commitish: pullRequest.head.ref,
+        previous_tag_name: previous_tag_name,
+      }
+    );
+
+    pullRequest.body = response.data.body;
+  } catch (error) {
+    pullRequest.body += "\nFailed to generate a body (" + error + ")";
+    throw Error("Failed to generate a body: " + error);
+  }
+}
+
 async function release(pullRequest) {
   core.info(`Releasing ${pullRequest.merge_commit_sha}..`);
   const tag = getTagName(pullRequest);
   const isPrerelease =
     JSON.parse(core.getInput("prerelease") || false) === true;
 
+  const shouldGenerateReleaseNotes =
+    JSON.parse(core.getInput("generate_release_notes") || false) === true;
+  if (shouldGenerateReleaseNotes) await generateReleaseNotes(pullRequest);
+
   core.info(`Is prerelease? ${isPrerelease}`);
+  core.info(`Is draft? ${shouldGenerateReleaseNotes}`);
   await client.rest.repos.createRelease({
     name: pullRequest.title,
     tag_name: tag,
+    draft: shouldGenerateReleaseNotes,
```

Comment:
Draft mode decision should not be based on if release notes are generated or not, it should probably be a separate input/setting.